pkg/kubectl/history.go

```diff
@@ -256,29 +256,50 @@ func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision
 	})
 }
 
-// controlledHistories returns all ControllerRevisions controlled by the given DaemonSet
-func controlledHistories(extensions clientextensionsv1beta1.ExtensionsV1beta1Interface, apps clientappsv1beta1.AppsV1beta1Interface, namespace, name string) (*extensionsv1beta1.DaemonSet, []*appsv1beta1.ControllerRevision, error) {
-	ds, err := extensions.DaemonSets(namespace).Get(name, metav1.GetOptions{})
+// controlledHistories returns all ControllerRevisions controlled by the given API object
+func controlledHistories(apps clientappsv1beta1.AppsV1beta1Interface, extensions clientextensionsv1beta1.ExtensionsV1beta1Interface, namespace, name, kind string) (runtime.Object, []*appsv1beta1.ControllerRevision, error) {
+	var obj runtime.Object
+	var labelSelector *metav1.LabelSelector
+
+	switch kind {
+	case "DaemonSet":
+		ds, err := extensions.DaemonSets(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
+		}
+		labelSelector = ds.Spec.Selector
+		obj = ds
+	case "StatefulSet":
+		ss, err := apps.StatefulSets(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to retrieve StatefulSet %s: %v", name, err)
+		}
+		labelSelector = ss.Spec.Selector
+		obj = ss
+	}
+
+	accessor, err := meta.Accessor(obj)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
+		return nil, nil, fmt.Errorf("failed to obtain accessor for %s named %s: %v", kind, accessor.GetName(), err)
 	}
+
 	var result []*appsv1beta1.ControllerRevision
-	selector, err := metav1.LabelSelectorAsSelector(ds.Spec.Selector)
+	selector, err := metav1.LabelSelectorAsSelector(labelSelector)
 	if err != nil {
 		return nil, nil, err
 	}
-	historyList, err := apps.ControllerRevisions(ds.Namespace).List(metav1.ListOptions{LabelSelector: selector.String()})
+	historyList, err := apps.ControllerRevisions(accessor.GetNamespace()).List(metav1.ListOptions{LabelSelector: selector.String()})
```

Comment:
I think they should be the same, since the logic chain is
parameters<name, namespace> -> obj -> accessor -> GetNamespace()